src/main/java/com/conveyal/gtfs/GeometryCache.java

```diff
@@ -0,0 +1,55 @@
+package com.conveyal.gtfs;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import com.github.benmanes.caffeine.cache.RemovalCause;
+import org.locationtech.jts.geom.Envelope;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.index.strtree.STRtree;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.function.BiConsumer;
+
+/**
+ * Cache sets of geometry in a JTS STRtree using a Caffeine LoadingCache.
+ */
+public class GeometryCache<T extends Geometry> {
```

Comment:
Should be renamed - it caches spatial indexes, not just geometries. Summarizing discussion: this may be "premature abstraction" (not optimization) since long term we're not certain to maintain two separate spatial index caches. Most reasons that would lead us to maintain two or more of them would also mean they potentially benefit from having different expiration and size eviction parameters. These hard-wired constants are not really constant across all potential spatial index caches, so this doesn't quite match the implied level of abstraction.

It's also not clear whether factoring this code out makes it easier or harder to read. If we retain this abstraction I'll update the Javadoc to include the above comments.